test/gtest/ucs/test_sock.cc

```diff
@@ -53,6 +69,48 @@ class test_socket : public ucs::test {
             EXPECT_EQ(exp_str, test_str);
         }
     }
+
+    void verify_subnet_match(sa_family_t af, const char *ip_addr1,
+                             const char *ip_addr2, unsigned max_common_prefix)
+    {
+        struct sockaddr_storage storage1 = {0};
+        struct sockaddr_storage storage2 = {0};
+        struct sockaddr *saddr1          = (struct sockaddr*)&storage1;
+        struct sockaddr *saddr2          = (struct sockaddr*)&storage2;
+        int matched;
+
+        fill_sockaddr(af, ip_addr1, 1234, saddr1);
+        fill_sockaddr(af, ip_addr2, 1234, saddr2);
+
+        static const size_t byte_to_bits = 8;
+        size_t addr_size_bits = get_addr_size(saddr1) * byte_to_bits;
+
+        for (size_t prefix = 0; prefix < addr_size_bits; ++prefix) {
```

Comment:
Seems that `prefix` is an index, while `max_common_prefix` is a length, so in case of 0 the result is wrong (0 <= 0 is true...)

Reply:
actually prefix is also length, and the result should be 'true' in that case. (prefix = 0 => always match).
but seems that the case where prefix == addr_size_bits is not covered by the loop.